File: backend/app/models/store.py
import uuid
from sqlalchemy import String, Double, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.database import Base


class Store(Base):
    __tablename__ = "stores"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    chain: Mapped[str] = mapped_column(String(50), nullable=False)   # 'loblaws', 'walmart', 'sobeys', 'metro'
    banner: Mapped[str | None] = mapped_column(String(100))           # 'No Frills', 'FreshCo', etc.
    name: Mapped[str] = mapped_column(Text, nullable=False)
    address: Mapped[str | None] = mapped_column(Text)
    city: Mapped[str | None] = mapped_column(String(100))
    province: Mapped[str | None] = mapped_column(String(2))
    postal_code: Mapped[str | None] = mapped_column(String(10))
    lat: Mapped[float | None] = mapped_column(Double)
    lng: Mapped[float | None] = mapped_column(Double)
    store_id: Mapped[str | None] = mapped_column(String(100), unique=True)  # retailer's own ID

    prices: Mapped[list["Price"]] = relationship("Price", back_populates="store")

File: mobile/lib/shared/models/store.dart
// ignore_for_file: invalid_annotation_target
import 'package:freezed_annotation/freezed_annotation.dart';

part 'store.freezed.dart';
part 'store.g.dart';

@freezed
class Store with _$Store {
  const factory Store({
    required String id,
    required String chain,
    String? banner,
    required String name,
    String? address,
    String? city,
    String? province,
    @JsonKey(name: 'postal_code') String? postalCode,
    double? lat,
    double? lng,
    @JsonKey(name: 'distance_km') double? distanceKm,
  }) = _Store;

  factory Store.fromJson(Map<String, dynamic> json) => _$StoreFromJson(json);
}